src/boot/efi/meson.build

```diff
@@ -266,11 +266,19 @@ efi_ldflags = [
         efi_crt0,
 ]
 
-possible_link_flags = [
-        '-Wl,--no-warn-execstack',
-        '-Wl,--no-warn-rwx-segments',
-]
-efi_ldflags += cc.get_supported_link_arguments(possible_link_flags)
+foreach arg : ['-Wl,--no-warn-execstack',
+               '-Wl,--no-warn-rwx-segments']
+        # We need to check the correct linker for supported args. This is what
+        # cc.has_multi_link_arguments() is for, but it helpfully overrides our
+        # choice of linker by putting its own -fuse-ld= arg after ours.
+        if run_command('bash', '-c',
```

Comment:
Purely cosmetic remark: maybe it would be good idea to follow convention for standard tools and obtain path to `bash` first: `bash = find_program('bash')`?

Likely could be placed in a top-level `meson.build` next to other standard tools:
https://github.com/systemd/systemd/blob/ff71363c26008a322afe70221dd4b03364b62614/meson.build#L653-L662

Since `bash` is already implicitly required by ie `generate-*` scripts:
https://github.com/systemd/systemd/blob/ff71363c26008a322afe70221dd4b03364b62614/src/basic/generate-af-list.sh#L1
https://github.com/systemd/systemd/blob/ff71363c26008a322afe70221dd4b03364b62614/src/basic/generate-arphrd-list.sh#L1
https://github.com/systemd/systemd/blob/ff71363c26008a322afe70221dd4b03364b62614/src/basic/generate-cap-list.sh#L1
https://github.com/systemd/systemd/blob/ff71363c26008a322afe70221dd4b03364b62614/src/basic/generate-errno-list.sh#L1

Reply:
It's not always done. And quite frankly, I fail to see the point of it. It amounts to the same thing with more work.